spec/requests/providers/about_financial_means_controller_spec.rb

```diff
@@ -0,0 +1,42 @@
+require "rails_helper"
+
+RSpec.describe Providers::AboutFinancialMeansController do
+  let(:legal_aid_application) { create(:legal_aid_application, :applicant_details_checked, applicant:) }
+  let(:provider) { legal_aid_application.provider }
+  let(:applicant) { create(:applicant) }
```

Comment:
As the applicant is not explicitly used, you could use the `:with_applicant` trait on the legal_aid_application?
```suggestion
  let(:legal_aid_application) { create(:legal_aid_application, :applicant_details_checked, :with_applicant) }
  let(:provider) { legal_aid_application.provider }
```